pkg/client/cache/delta_fifo.go

```diff
@@ -452,6 +460,16 @@ func (f *DeltaFIFO) Replace(list []interface{}, resourceVersion string) error {
 	return nil
 }
 
+// Resync will send a sync event for each item
+func (f *DeltaFIFO) Resync() error {
+	f.lock.RLock()
+	defer f.lock.RUnlock()
+	if err := f.replaceLocked(f.listLocked()); err != nil {
```

Comment:
Resync cannot do deletion detection, because it's comparing the content of store with itself.

I didn't look into TestUpdate, but it seems we probably need to change it.